inginious/frontend/courses.py

```diff
@@ -215,12 +216,35 @@ def get_all_tags_names_as_list(self, admin=False, language="en"):
             return self._all_tags_cache_list_admin[language]
         return self._all_tags_cache_list[language]
         
+    def get_organisational_tags_to_task(self):
+        """ This build a dict for fast retrive tasks id based on organisational tags. The form of the dict is:
+        
+            { 'org_tag_1': ['task_id', 'task_id', ...], 
+              'org_tag_2' : ['task_id', 'task_id', ...], 
+              ... }
+         """
+        if self._organisational_tags_to_task != {}:
+            return self._organisational_tags_to_task
+            
+        for org_tag in self.get_all_tags()[2]:
+            tag = org_tag.get_name()
```

Comment:
Is this gettext friendly ?

Reply:
Yes it should, the translations are performed later in the HTML (submission_viewer.html, line 93).